### PR TITLE
docs: mention pre-commit hook

### LINKED ISSUE #8: Usage from pre-commit
Currently, this works perfectly like this:

```yaml
- id: clang-format
  name: Clang Format
  language: python
  types:
  - c++
  entry: clang-format
  additional_dependencies: [clang-format==12.*]
  args:
  - -style=file
  - -i
```

However, this will not get update PRs from pre-commit.ci, or support `pre-commit autoupdate`. Ideally, it would be nice to have a hook repository for this. The "obvious" thing, adding a `.pre-commit-hooks.yml` to this repository, would not work, since it will try to build from source. The ideal thing is likely to follow the method used by https://github.com/cheshirekow/cmake-format-precommit, and have a clang-format-precommit repository. It would be very simple, but would require a bit of maintenance every time there's a new release (update a line, make a tag/release). Would you be interested if I set up such a repository and then moved it here? Or I could set this up elsewhere (like in pybind).

### PR DESCRIPTION
**Pull request by henryiii**
Closes #8.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,17 @@ python -m pip install clang-format
...
 
+## Use from pre-commit
+
+A [pre-commit](https://pre-commit.com) hook is also [provided](https://github.com/ssciwr/clang-format-precommit), use like this:
+
+```yaml
+- repo: https://github.com/ssciwr/clang-format-precommit
+  rev: v12.0.1
+  hooks:
+  - pre-commit
```
Comment (henryiii) [inline README.md:21]:
```suggestion
  - id: pre-commit
```

File: README.md [hunk 2]
```diff
@@ -10,6 +10,17 @@ python -m pip install clang-format
...
 
+## Use from pre-commit
+
+A [pre-commit](https://pre-commit.com) hook is also [provided](https://github.com/ssciwr/clang-format-precommit), use like this:
+
+```yaml
+- repo: https://github.com/ssciwr/clang-format-precommit
+  rev: v12.0.1
+  hooks:
+  - id: pre-commit
```
Comment (dokempf) [inline README.md:21]: It's `id: clang-format` - isn't it?

Comment (henryiii) [reply README.md:21]: Ahhhh, yes. 🤦

**Comment by henryiii** [comment]
FYI, you can see the new hook in action here: https://github.com/scikit-hep/boost-histogram/pull/610/files

**Pull request merged**